model-optimizer/extensions/middle/ONNXResize11ToInterpolateV4.py

```diff
@@ -0,0 +1,174 @@
+
+    shape_calculation_mode = 'scales' if num_of_inputs == 3 else 'sizes'
+
+    interpolate_node = Interpolate(graph, {'version': 'opset4',
+                                           'mode': convert_mode(resize.mode),
+                                           'coordinate_transformation_mode': resize.coordinate_transformation_mode,
+                                           'cube_coeff': resize.cube_coeff,
+                                           'nearest_mode': resize.nearest_mode,
+                                           'pads_begin': int64_array([0]),
+                                           'pads_end': int64_array([0]),
+                                           'antialias': 0,
+                                           'shape_calculation_mode': shape_calculation_mode,
+                                           'in_ports_count': 4}).create_node()
+
+    axes_node.out_port(0).connect(interpolate_node.in_port(3))
+    shape_of = Shape(graph, {'name': resize_name + '/ShapeOf_'}).create_node()
+
+    add_node = create_op_with_const_inputs(graph, Add,
+                                           {1: float_array([1.0e-5])},
+                                           {'name': resize_name + '/Add_'})
+
+    if num_of_inputs == 3:
+        input_data_type = data_type_str_to_np(graph.graph['cmd_params'].data_type)
+        cast_shape_to_float = Cast(graph, {'dst_type': input_data_type}).create_node()
+        mul_node = Mul(graph, {'name': resize_name + '/Mul_'}).create_node()
+        shape_of.out_port(0).connect(cast_shape_to_float.in_port(0))
+        cast_shape_to_float.out_port(0).connect(mul_node.in_port(0))
+        cast_add_result_to_int = Cast(graph, {'dst_type': np.int64}).create_node()
+        floor_node = Floor(graph, {'name': resize_name + '/Floor_'}).create_node()
+        mul_node.out_port(0).connect(add_node.in_port(0))
+        add_node.out_port(0).connect(floor_node.in_port(0))
+        floor_node.out_port(0).connect(cast_add_result_to_int.in_port(0))
+        cast_add_result_to_int.out_port(0).connect(sizes_ss.in_port(0))
+        sizes_ss.out_port(0).connect(interpolate_node.in_port(1))
+        scales_ss.out_port(0).connect(interpolate_node.in_port(2))
+
+        rename_nodes([(resize, resize_name + '/delete'), (interpolate_node, resize_name)])
+
+        connection_of_resize_input = resize.in_port(0).get_connection()
+        connection_of_resize_input.set_destination(interpolate_node.in_port(0))
+
+        connection_of_scales = resize.in_port(2).get_connection()
+        connection_of_scales.set_destination(scales_ss.in_port(0))
+
+        connection_of_resize_input.get_source().connect(shape_of.in_port(0))
+        connection_of_scales.get_source().connect(mul_node.in_port(1))
+    else:
+        cast_shape_to_float = Cast(graph, {'dst_type': np.float32}).create_node()
+        cast_sizes_to_float = Cast(graph, {'dst_type': np.float32}).create_node()
```

Comment:
Change `np.float32` to `data_type_str_to_np(graph.graph['cmd_params'].data_type)`. Move variable `input_data_type` out of `if` statement from line 110.

Reply:
Done.